inc/cli/class-assign-defaults-command.php

```diff
@@ -0,0 +1,240 @@
+<?php
+/**
+ * HM Utility Taxonomy CLI: Assign default terms to existing published posts
+ */
+
+declare( strict_types=1 );
+
+namespace HM\Utility_Taxonomy\CLI;
+
+use HM\Utility_Taxonomy as HMUT;
+use WP_CLI;
+use WP_Post;
+use WP_Query;
+
+/**
+ * Assign default terms
+ *
+ * @since 1.4.0
+ */
+class Assign_Defaults_Command {
+	/**
+	 * Number of found posts
+	 *
+	 * @since 1.4.0
+	 *
+	 * @var int
+	 */
+	protected $found_posts = 0;
+
+	/**
+	 * Maximum number of pages
+	 *
+	 * @since 1.4.0
+	 *
+	 * @var int
+	 */
+	protected $max_pages = 0;
+
+	/**
+	 * Current paged number
+	 *
+	 * @since 1.4.0
+	 *
+	 * @var int
+	 */
+	protected $paged = 1;
+
+	/**
+	 * Default posts per page
+	 *
+	 * @since 1.4.0
+	 *
+	 * @var int
+	 */
+	protected $posts_per_page = 100;
+
+	/**
+	 * Used to check whether we're dry-running or not
+	 *
+	 * @since 1.4.0
+	 *
+	 * @var bool
+	 */
+	protected $is_dry_running = false;
+
+	/**
+	 * Map term IDs to term names
+	 *
+	 * @since 1.4.0
+	 *
+	 * @param array $term_ids Array of term IDs.
+	 *
+	 * @return array
+	 */
+	protected function term_ids_to_names( array $term_ids ) : array {
+		return array_map( function ( int $id ) : string {
+			return get_term( $id, HMUT\TAXONOMY )->name;
+		}, $term_ids );
+	}
+
+	/**
+	 * Process post
+	 *
+	 * @since 1.4.0
+	 *
+	 * @param WP_Post $post Post object.
+	 *
+	 * @return void
+	 */
+	protected function process( WP_Post $post ) {
+		$term_ids = HMUT\get_post_default_term_ids( $post->ID );
+
+		if ( empty( $term_ids ) ) {
+			WP_CLI::log( sprintf( 'No default terms found for %d.', $post->ID ) );
+			return;
+		}
+
+		if ( ! $this->is_dry_running ) {
+			wp_set_post_terms( $post->ID, $term_ids, HMUT\TAXONOMY, true );
+		}
+
+		WP_CLI::log(
+			sprintf(
+				'Setting default terms of post %d: %s.',
+				$post->ID,
+				join( ', ', $this->term_ids_to_names( $term_ids ) )
+			)
+		);
+	}
+
+	/**
+	 * Get query
+	 *
+	 * @since 1.4.0
+	 *
+	 * @param array $args Array of query arguments to pass to WP_Query.
+	 *
+	 * @return WP_Query
+	 */
+	protected function get_query( array $args ) : WP_Query {
+		$query_args = wp_parse_args( $args, [
+			'paged'          => $this->paged,
+			'post_status'    => 'publish',
```

Comment:
Depending on the usecase, only updating published posts would not be desired, right? Although it's fine for the case this PR was created to address :D 

However it seems to me that the example later wouldn't work because of this?
`wp hm-utility-taxonomy assign-defaults --post_type=page,post --post_status=draft`

I haven't been able to test, so could be wrong.

Reply:
It should work because the default arguments are overriden by the passed one.